### PR TITLE
Implement progress bars for the etcher writer

### LINKED ISSUE #113: Show progress bar while etcher flashes image

### PR DESCRIPTION
**Pull request by telphan**
Closes #113 
Change-type: patch
Signed-off-by: Theodor Gherzan <theodor@resin.io>

### CODE REVIEW

**Comment by telphan** [comment]
@jviotti Ping this for an approve.

File: lib/components/resinio/sdk.js [hunk 1]
```diff
@@ -40,16 +40,27 @@ exports.getAllSupportedOSVersions = (deviceType) => {
...
       bar.update({
         percentage: data.percentage,
         eta: data.eta
       })
-    })
+    }
   }
 
+  stream.on('progress', (data) => {
+    report(data)
```
Comment (jviotti) [inline lib/components/resinio/sdk.js]: Could we put the different report logic depending on `process.env.CI` inside here?

File: lib/components/writer/etcher.js [hunk 2]
```diff
@@ -48,13 +49,33 @@ exports.writeImage = async (image, destination) => {
     check: true
   })
 
-  emitter.on('progress', (state) => {
-    console.log(`Flashing ${image}: ${state.percentage.toFixed(2)}% (${state.type})`)
+  const bar = new visuals.Progress('')
+
+  let report = (data) => {
+    if (Math.floor(data.percentage) % 5 === 0) {
```
Comment (jviotti) [inline lib/components/writer/etcher.js]: So this is basically the same logic we have on the SDK. Should we create shared utility function to "print" progress, that decides what to do depending on the `CI` variable?

**Comment by horia-delicoti** [comment]
Fixed in #258

**Pull request closed without merging**